### ISSUE TITLE
Cannot gracefully handle error

### ISSUE THREAD

**Issue by ComputerCarl** [axios/axios#1533]
#### Summary

When this page *redirects*, it crashes the application and no error is caught.
```js
const axios = require('axios');
const debug = require('debug')('downloadToBuffer');

// certificate errors some axios gets
// https://stackoverflow.com/questions/31673587/error-unable-to-verify-the-first-certificate-in-nodejs
// https://github.com/axios/axios/issues/284
// https://github.com/axios/axios/issues/535

var rootCas = require('ssl-root-cas').create();
require('https').globalAgent.options.ca = rootCas;

const https = require('https');
const agent = new https.Agent({ rejectUnauthorized: false });

let url = 'https://ih1.redbubble.net/image.99400032.8054/flat';
axios.get(url, {
    responseType: 'arraybuffer',
    httpsAgent: agent
})
    .then(result => ({ buffer: result.data }))
    .catch(err => {
        debug({ err });
    });
```

produces this;
```
    _http_client.js:131
    throw new Error('Protocol "' + protocol + '" not supported. ' +
    ^

Error: Protocol "http:" not supported. Expected "https:"
    at new ClientRequest (_http_client.js:131:11)
    at Object.request (http.js:38:10)
    at RedirectableRequest._performRequest (...\node_modules\follow-redirects\index.js:128:24)
    at RedirectableRequest._processResponse (...\node_modules\follow-redirects\index.js:209:10)
    at ClientRequest.RedirectableRequest._onNativeResponse (...\node_modules\follow-redirects\index.js:38:10)
    at Object.onceWrapper (events.js:315:30)
    at emitOne (events.js:116:13)
    at ClientRequest.emit (events.js:211:7)
    at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:551:21)
    at HTTPParser.parserOnHeadersComplete (_http_common.js:116:23)
    at TLSSocket.socketOnData (_http_client.js:440:20)
    at emitOne (events.js:116:13)
    at TLSSocket.emit (events.js:211:7)
    at addChunk (_stream_readable.js:263:12)
    at readableAddChunk (_stream_readable.js:250:11)
    at TLSSocket.Readable.push (_stream_readable.js:208:10)
```

#### Context

- axios version:  v0.18.0
- Environment: node v8.9.4, firefox, windows 7

**Comment by yadue** [comment]
Same error for me, cant catch that error, breaks node js server.

**Comment by Kelanor** [comment]
Same here too. I can't catch this error. It just crashes node application

**Comment by PavelPolyakov** [comment]
@faburu do you have `https_proxy` env variable set when executing this?

I have exactly the same error, but mine is about `https_proxy`. Here is the issue: https://github.com/axios/axios/issues/1885

**Comment by chinesedfan** [comment]
Closed in favor of #1885.
